src/ucp/rndv/rndv_rtr.c

```diff
@@ -180,14 +180,16 @@ static ucs_status_t ucp_proto_rndv_rtr_progress(uct_pending_req_t *self)
     ucs_status_t status;
 
     if (!(req->flags & UCP_REQUEST_FLAG_PROTO_INITIALIZED)) {
-        status = ucp_datatype_iter_mem_reg(req->send.ep->worker->context,
-                                           &req->send.state.dt_iter,
-                                           rpriv->super.md_map,
-                                           UCT_MD_MEM_ACCESS_REMOTE_PUT,
-                                           UCS_BIT(UCP_DATATYPE_CONTIG));
-        if (status != UCS_OK) {
-            ucp_proto_request_abort(req, status);
-            return UCS_OK;
+        if (rpriv->super.md_map != 0) {
```

Comment:
why is it not covered by ucp_datatype_iter_mem_reg already?

Reply:
because I removed a branch from there according to the previous comment.
The failing flow is ud + contig datatype. If we get an empty memh here ,we will need to put it back later, which is not done in case of `receive_bcopy` (when we get messages by the AMs)